cmake/kokkos_functions.cmake

```diff
@@ -60,6 +60,278 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
 
 ENDFUNCTION()
 
+FUNCTION(kokkos_append_config_line LINE)
+  GLOBAL_APPEND(KOKKOS_TPL_EXPORTS "${LINE}")
+ENDFUNCTION()
+
+MACRO(kokkos_export_cmake_tpl NAME)
+  KOKKOS_APPEND_CONFIG_LINE("IF(NOT DEFINED ${NAME}_DIR AND NOT DEFINED ${NAME}_ROOT)")
+  KOKKOS_APPEND_CONFIG_LINE("  SET(${NAME}_DIR  ${${NAME}_DIR})")
+  KOKKOS_APPEND_CONFIG_LINE("  SET(${NAME}_ROOT ${${NAME}_ROOT})")
```

Comment:
This currently defines empty variables if `${NAME}_DIR` is set while configuring but `${NAME}_ROOT` is not. I guess you only want to set the values (individually) if they were found at configure time.

Reply:
Changed.  It's a bit funky since if you give `${NAME}_ROOT` then `${NAME}_DIR` automatically gets defined.